apps/api/src/domain/models/vocabulary-list/entities/vocabulary-list.entity.ts

```diff
@@ -268,6 +276,78 @@ export class VocabularyList extends Resource {
         return this;
     }
 
+    @UpdateMethod()
+    importEntries(importItems: VocabularyListEntryImportItem[]): ResultOrError<VocabularyList> {
+        if (this.entries.length > 0) {
+            return new FailedToImportEntriesToVocabularyListError(this.id, [
+                new InternalError(
+                    `you cannot import entries to a vocabulary list that already has entries`
+                ),
+            ]);
+        }
+
+        if (importItems.length === 0) {
+            return new FailedToImportEntriesToVocabularyListError(this.id, [
+                new InternalError(
+                    `you must provide at least one item when importing entries to a vocabulary list`
+                ),
+            ]);
+        }
+
+        const unknownFilterPropertyErrors = importItems
+            .filter(({ propertyValues }) =>
+                Object.keys(propertyValues).some((key) => !this.hasFilterPropertyNamed(key))
+            )
+            .map(
+                ({ termId }) =>
+                    new InternalError(
+                        `failed to import an entry for ${termId} as one of the provided properties has not been registered`
+                    )
+            );
+
+        if (unknownFilterPropertyErrors.length > 0) {
+            return new FailedToImportEntriesToVocabularyListError(
+                this.id,
+                unknownFilterPropertyErrors
+            );
+        }
+
+        const unknownPropertyValueErrors = importItems
```

Comment:
We might want to update this to be able to say in the error message which property values are not allowed.

Reply:
Done.